apps/crud/src/app/todos/todo.store.ts

```diff
@@ -0,0 +1,117 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Injectable, inject } from '@angular/core';
+import {
+  ComponentStore,
+  OnStateInit,
+  OnStoreInit,
+  tapResponse,
+} from '@ngrx/component-store';
+import { pipe, switchMap, tap } from 'rxjs';
+import { TodoService } from './todos.service';
+import { Todo } from './todo.model';
+interface TodoState {
+  todos: Todo[];
+  loading: boolean;
+  error: HttpErrorResponse | null;
+}
+
+const initState: TodoState = { todos: [], loading: false, error: null };
+
+@Injectable()
+export class TodoStore
+  extends ComponentStore<TodoState>
+  implements OnStateInit, OnStoreInit
+{
+  private readonly todoService = inject(TodoService);
+  #todos$ = this.select((s) => s.todos);
+  #loading$ = this.select((s) => s.loading);
+  #error$ = this.select((s) => s.error);
+  readonly vm$ = this.select(
+    {
+      todos: this.#todos$,
+      loading: this.#loading$,
+      error: this.#error$,
+    },
+    { debounce: true }
+  );
+
+  // private methods
```

Comment:
personal preference: I'm not a big fan of useless comment. 

using the # should be enough to know that the property is private.